planner/core/task.go

```diff
@@ -1341,10 +1345,14 @@ func (p *basePhysicalAgg) convertAvgForMPP() *PhysicalProjection {
 			newAggFuncs = append(newAggFuncs, &avgSum)
 			newSchema.Append(p.schema.Columns[i])
 			avgSumCol := p.schema.Columns[i]
-			// if(avgCountCol = 0, NULL, avgSumCol/avgCountCol)
+			// if(avgCountCol = 0, NULL, avgSumCol/avgCountCol), but because Clickhouse has a bug which evaluates every branch, so
+			// if(avgCountCol = 0, NULL, avgSumCol/(case when avgCountCol=0 then 1 else avgCountCol end))
```

Comment:
So we don't really need the if condition right? Just sumCol / if(Count = 0, 1, Count) would suffice.

Reply:
nice!